fingerprint/infra/bio-sdk/src/main/java/com/simprints/fingerprint/infra/biosdk/FingerprintBioSdkModule.kt

```diff
@@ -1,17 +1,54 @@
 package com.simprints.fingerprint.infra.biosdk
 
+import com.simprints.fingerprint.infra.biosdkimpl.SimprintsSdk
+import com.simprints.fingerprint.infra.necsdkimpl.NecSdk
+import com.simprints.infra.config.store.ConfigRepository
+import com.simprints.infra.config.store.models.FingerprintConfiguration
 import dagger.Binds
 import dagger.Module
+import dagger.Provides
 import dagger.hilt.InstallIn
 import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.runBlocking
+import javax.inject.Singleton
 
 @Module
 @InstallIn(SingletonComponent::class)
 internal abstract class FingerprintBioSdkModule {
 
-    // TODO Consider replacing this with a provider method that would inject
-    //  specific wrappers based on project configuration.
+    @Binds
+    @SimprintsSdk
+    @Singleton
+    abstract fun provideSimprintsBioSdkWrapper(impl: SimprintsBioSdkWrapper): BioSdkWrapper
 
     @Binds
-    abstract fun provideBioSdkWrapper(impl: SimprintsBioSdkWrapper): BioSdkWrapper
+    @NecSdk
+    @Singleton
+    abstract fun provideNecBioSdkWrapper(impl: NECBioSdkWrapper): BioSdkWrapper
+}
+
+@Module
+@InstallIn(SingletonComponent::class)
+internal object FingerprintBioSdkProvidersModule {
+
+
+    @Provides
+    @Singleton
+    fun provideBioSdkWrapper(
+        @SimprintsSdk simprintsWrapper: BioSdkWrapper?,
+        @NecSdk necWrapper: BioSdkWrapper?,
+        configManager: ConfigRepository
+    ): BioSdkWrapper = runBlocking {
+        // Todo we didn't yet implement the logic to select the SDK based on the configuration
+        // so we are just using the first allowed SDK for now
+        // See tickets in SIM-81 for more details
+        when (configManager.getProjectConfiguration().fingerprint?.allowedSDKs?.first()) {
```

Comment:
I am not sure it is a good idea to do disk reads (and potentially network calls) in the dependency resolution. I suggest creating `BioSdkResolverUseCase` that could be injected instead of the wrapper where needed and resolve which SDK to return on call time instead of the DI resolution time.

```
class BioSdkResolverUseCase @Inject constructor(
  @SimprintsSdk private val simprintsWrapper: BioSdkWrapper,
  @NecSdk private val necWrapper: BioSdkWrapper,
  private val configManager: ConfigRepository,
) {
  operator suspend fun invoke(): BioSdkWrapper = ...
}
```